pkg/remote/aws/ec2_network_acl_enumerator.go

```diff
@@ -0,0 +1,50 @@
+package aws
+
+import (
+	"github.com/cloudskiff/driftctl/pkg/remote/aws/repository"
+	remoteerror "github.com/cloudskiff/driftctl/pkg/remote/error"
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+)
+
+type EC2NetworkACLEnumerator struct {
+	repository repository.EC2Repository
+	factory    resource.ResourceFactory
+}
+
+func NewEC2NetworkACLEnumerator(repo repository.EC2Repository, factory resource.ResourceFactory) *EC2NetworkACLEnumerator {
+	return &EC2NetworkACLEnumerator{
+		repository: repo,
+		factory:    factory,
+	}
+}
+
+func (e *EC2NetworkACLEnumerator) SupportedType() resource.ResourceType {
+	return aws.AwsNetworkACLResourceType
+}
+
+func (e *EC2NetworkACLEnumerator) Enumerate() ([]*resource.Resource, error) {
+	resources, err := e.repository.ListAllNetworkACLs()
+	if err != nil {
+		return nil, remoteerror.NewResourceListingError(err, string(e.SupportedType()))
+	}
+
+	results := make([]*resource.Resource, 0, len(resources))
+
+	for _, res := range resources {
+		// Do not handle default network acl since it is a dedicated resource
+		if *res.IsDefault {
+			continue
+		}
```

Comment:
Yeah I think it was a mistake to make `ListAllVPCs` aware of the default concept, this is not the role of a repository. IMHO  BTW the way we use two cache key in this method is weird. There is only one thing that we want to cache : aws sdk call result, it does not make a lot of sense to cache these two results. I mean the cost of iterating the slice and filter default vpc is insignificant compared to a network call ... I also like the idea to have a single enumerator per resource type, the way we did VPC was an anti-pattern for me, maybe we do it like this before we introduced the repository pattern and it was to avoid doing the API call twice ?

Reply:
Yep we did create the repository pattern after. Ok I'm good with your way of only having one enumerator for 2 "identical" resources (e.g. default and non-default)